### ISSUE TITLE
Reproducing the results on RealSR

### ISSUE THREAD

**Issue by LearningHx** [NJU-PCALab/AddSR#10]
Hello, I am trying to reproduce the results of your paper on RealSR, but I have found that the metrics I tested differ from those in your paper (I have used the pre-trained model and dataset you provided).  In addition, I find that there are differences in the metrics reported in the different versions of your paper, and I'm wondering what causes this.

![图片2](https://github.com/NJU-PCALab/AddSR/assets/109660296/cbc1c29a-1fc7-425f-9af3-696cd382a2eb)

**Comment by CSRuiXie** [comment]
Thanks for your attention to our work!
1. The difference between AddSR(v1) and AddSR may be due to the usage of PSR. In the v1 paper, we used predicted HR to control the model output. However, we found that using the interpolation of LR and predicted HR achieves better results. Therefore, we use the latter method to control the model instead of relying solely on predicted HR.
2. The difference between AddSR(v1) and AddSR(v3) may be due to the different forms of TA-ADD. In the v1 paper, we used a simple manually defined piecewise function. However, we found that using an exponential form as the weighting function is better for four steps, so we used the latter loss function.

**Comment by DDDaxing** [comment]
> Thanks for your attention to our work!
> 
> 1. The difference between AddSR(v1) and AddSR may be due to the usage of PSR. In the v1 paper, we used predicted HR to control the model output. However, we found that using the interpolation of LR and predicted HR achieves better results. Therefore, we use the latter method to control the model instead of relying solely on predicted HR.
> 2. The difference between AddSR(v1) and AddSR(v3) may be due to the different forms of TA-ADD. In the v1 paper, we used a simple manually defined piecewise function. However, we found that using an exponential form as the weighting function is better for four steps, so we used the latter loss function.

Hi authors,
If I'd like to fine-tune your addsr model, should I use the interpolation of LR and predicted HR during training? Since I found that in the training script, you only use the predicted HR. 
BTW, why do you add noise to the latents after each step in inference stage? I just don't understand. Can you please kindly clarify the reason? Thank you.

**Comment by CSRuiXie** [comment]
Thanks for your attention to our work!

   1.

   You can use interpolation between LR and predicted HR images during
   training to enhance the fidelity of the restored results. Although we
   haven’t updated the code in the repository recently due to my busy
   schedule, you can modify it easily to include this interpolation.
   2.

   In the inference pipeline, we predict x0 at each step. Therefore, we
   need to add noise to x0 to diffuse it to the next noise state, which
   serves as the input for the subsequent step.


T W ***@***.***> 于2024年9月4日周三 10:53写道：
